torch/csrc/jit/python/pybind_utils.h

```diff
@@ -256,8 +256,7 @@ InferredType tryToInferContainerType(py::handle input);
 inline InferredType tryToInferType(py::handle input) {
   // Try tensor types
   if (THPVariable_Check(input.ptr())) {
-    auto tensor = py::cast<at::Tensor>(input);
-    return InferredType(TensorType::create(tensor));
+    return InferredType(TensorType::get());
```

Comment:
TensorType::getInferred()?

Reply:
TensorType::getInferred is a typing-level construct, which says that an unannotated variable is being inferred to be a Tensor. IMO this doesn't have a lot to do with using a `Tensor` for an `Any` arg